golem-test-framework/src/components/component_compilation_service/mod.rs

```diff
@@ -0,0 +1,121 @@
+    }
+
+    async fn enqueue_compilation(&self, component_id: &ComponentId, component_version: u64) {
+        let response = self
+            .client()
+            .await
+            .enqueue_compilation(ComponentCompilationRequest {
+                component_id: Some(component_id.clone().into()),
+                component_version,
+            })
+            .await
+            .expect("Failed to enqueue component compilation")
+            .into_inner();
+        match response.result {
+            None => {
+                panic!("Missing response from golem-component-service for component compilation")
+            }
+            Some(component_compilation_response::Result::Success(_)) => (),
+            Some(component_compilation_response::Result::Failure(error)) => {
+                panic!("Failed to enqueue component compilation in golem-component-compilation-service: {error:?}");
+            }
+        }
+    }
+
+    fn private_host(&self) -> String;
+    fn private_http_port(&self) -> u16;
+    fn private_grpc_port(&self) -> u16;
+
+    fn public_host(&self) -> String {
+        self.private_host()
+    }
+
+    fn public_http_port(&self) -> u16 {
+        self.private_http_port()
+    }
+
+    fn public_grpc_port(&self) -> u16 {
+        self.private_grpc_port()
+    }
+
+    fn kill(&self);
+}
+
+async fn new_client(host: &str, grpc_port: u16) -> ComponentCompilationServiceClient<Channel> {
+    ComponentCompilationServiceClient::connect(format!("http://{host}:{grpc_port}"))
+        .await
+        .expect("Failed to connect to golem-component-compilation-service")
+}
+
+async fn wait_for_startup(host: &str, grpc_port: u16, timeout: Duration) {
+    wait_for_startup_grpc(
+        host,
+        grpc_port,
+        "golem-component-compilation-service",
+        timeout,
+    )
+    .await
+}
+
+fn env_vars(
+    http_port: u16,
+    grpc_port: u16,
+    component_service: Arc<dyn ComponentService + Send + Sync + 'static>,
+    verbosity: Level,
+) -> HashMap<String, String> {
+    let log_level = verbosity.as_str().to_lowercase();
+
+    let vars: &[(&str, &str)] = &[
+        ("RUST_LOG", &format!("{log_level},cranelift_codegen=warn,wasmtime_cranelift=warn,wasmtime_jit=warn,h2=warn,hyper=warn,tower=warn")),
+        ("RUST_BACKTRACE", "1"),
+        ("GOLEM__COMPILED_COMPONENT_SERVICE__TYPE", "Local"),
+        ("GOLEM__COMPILED_COMPONENT_SERVICE__CONFIG__ROOT", "/tmp/ittest-local-object-store/golem"),
+        ("GOLEM__COMPONENT_SERVICE__ACCESS_TOKEN", "2A354594-7A63-4091-A46B-CC58D379F677"),
+        ("GOLEM__COMPONENT_SERVICE__HOST", &component_service.private_host()),
+        ("GOLEM__COMPONENT_SERVICE__PORT", &component_service.private_grpc_port().to_string()),
+        ("GOLEM__GRPC_PORT", &grpc_port.to_string()),
+        ("GOLEM__HTTP_PORT", &http_port.to_string()),
+    ];
+
+    HashMap::from_iter(vars.iter().map(|(k, v)| (k.to_string(), v.to_string())))
+}
```

Comment:
I am slightly concerned about this repetition of configuration values.
I raised this, but I am sure, team has moved on